### PR TITLE
Golang policy controller docs

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

Docs changes for golang policy controller.  Mostly this is just removing the API access configuration from docs / manifests since the go policy controller has better support for in-cluster-config, eliminating the need for user configured API access. 

## Todos

- [x] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
The policy controller options `CONFIGURE_ETC_HOSTS` and `K8S_API` are no longer supported.  If needed, use `KUBECONFIG` instead.
```

### CODE REVIEW

File: master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml [hunk 1]
```diff
@@ -303,23 +303,15 @@ spec:
       serviceAccountName: calico-policy-controller
       containers:
         - name: calico-policy-controller
-          image: quay.io/calico/kube-policy-controller:{{site.data.versions[page.version].first.components["calico/kube-policy-controller"].version}}
+          # image: quay.io/calico/kube-policy-controller:{{site.data.versions[page.version].first.components["calico/kube-policy-controller"].version}}
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml]: Need to revert this change.

Comment (caseydavenport) [reply master/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml]: Also need to update other etcd manifests

File: master/reference/policy-controller/configuration.md [hunk 2]
```diff
@@ -13,57 +13,35 @@ section.
...
+| ETCD_ENDPOINTS    | The list of etcd nodes in your cluster. e.g `http://10.0.0.1:2379,http://10.0.0.2:2379`
+| ETCD_CA_CERT_FILE | The full path to the CA certificate file for the Certificate Authority that signed the etcd server key/certificate pair. | path
+| ETCD_CERT_FILE    | The full path to the client certificate file for accessing the etcd cluster. | path
+| ETCD_KEY_FILE     | The full path to the client key file for accessing the etcd cluster. | path
 
-> **Note**: When running etcd with TLS enabled, the addresses in 
-> `ETCD_ENDPOINTS` must be hostname values such as `etcd-host:2379`, 
-> NOT IP addresses.
-{: .alert .alert-info}
+The `*_FILE` variables are _paths_ to the corresponding certificates / keys. As such, when the policy controller is running as a Kubernetes pod, you
```
Comment (emanic) [inline master/reference/policy-controller/configuration.md]: I think that the additional spaces around the `/` are not necessary. I would delete those so that it read `certificates/keys`.

File: master/reference/policy-controller/configuration.md [hunk 3]
```diff
@@ -13,57 +13,35 @@ section.
...
+must ensure that the files exist within the pod. This is usually done in one of two ways:
 
-The `*_FILE` variables are _paths_ to the corresponding certificates / keys.  As such, when the policy controller is running as a Kubernetes pod, you
-must ensure that the files exist within the pod.  This is usually done in one of two ways:
-
-* Mount the certificates from the host.  This requires that the certs be present on the host that the policy controller is scheduled to / running on.
+* Mount the certificates from the host. This requires that the certs be present on the host that the policy controller is scheduled to / running on.
```
Comment (emanic) [inline master/reference/policy-controller/configuration.md]: I would suggest spelling out certs in the second sentence (as it is in the first sentence), so it would be certs --> certificates in sentence 2.

Comment (emanic) [reply master/reference/policy-controller/configuration.md]: For me, it is a little confusing to equate "scheduled to" with "running on". I guess that is primarily because "scheduled to" indicates something that will happen in the future and "running on" indicates the opposite. Also, I often see these more collapsed as "scheduled to run on." So, I personally cannot quite parse this construction but perhaps it's because I am relatively new to Kubernetes.

Comment (caseydavenport) [reply master/reference/policy-controller/configuration.md]: Yep, I agree.  I've re-worded as "This requires that the certs be present on the host running the policy controller"
